src/screens/SignTransactionSheet.tsx

```diff
@@ -107,14 +107,31 @@ export const SignTransactionSheet = () => {
 
   const addressToUse = specifiedAddress ?? accountAddress;
 
-  const { provider, nativeAsset } = useProviderSetup(chainId, addressToUse);
+  const provider = getProvider({ chainId });
+  const [nativeAsset, setNativeAsset] = useState<ParsedAddressAsset | null>(null);
 
   const isMessageRequest = isMessageDisplayType(transactionDetails.payload.method);
   const isPersonalSignRequest = isPersonalSign(transactionDetails.payload.method);
 
   const label = useForegroundColor('label');
   const surfacePrimary = useBackgroundColor('surfacePrimary');
 
+  useEffect(() => {
```

Comment:
we have `useNativeAsset({ chainId })` btw

Reply:
very nice - thank you!